lxc/config.go

```diff
@@ -308,6 +333,51 @@ func doProfileCreate(client *lxd.Client, p string) error {
 	return err
 }
 
+func doProfileEdit(client *lxd.Client, p string) error {
+	profile, err := client.ProfileConfig(p)
+	editor := os.Getenv("VISUAL")
+	if editor == "" {
+		editor = os.Getenv("EDITOR")
+		if editor == "" {
+			editor = "/usr/bin/vi"
+		}
+	}
+	data, err := yaml.Marshal(&profile)
+	f, err := ioutil.TempFile("/tmp", "image_")
```

Comment:
On Mon, Mar 09, 2015 at 08:44:17AM -0700, St�phane Graber wrote:
> > @@ -308,6 +333,51 @@ func doProfileCreate(client *lxd.Client, p string) error {
> >  	return err
> >  }
> >  
> > +func doProfileEdit(client *lxd.Client, p string) error {
> > +	profile, err := client.ProfileConfig(p)
> > +	editor := os.Getenv("VISUAL")
> > +	if editor == "" {
> > +		editor = os.Getenv("EDITOR")
> > +		if editor == "" {
> > +			editor = "/usr/bin/vi"
> > +		}
> > +	}
> > +	data, err := yaml.Marshal(&profile)
> > +	f, err := ioutil.TempFile("/tmp", "image_")
> 
> Should we perhaps include "lxc" in the tempfile prefix here?

Not sure, but we should probably (?) remove "/tmp" and just make it
""; golang is smart enough to respect environment variables and do the
right thing on different platforms, so if we ever want to put
temporary files somewhere else, we'd have to change the code.

> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/345/files#r26046693